File: CodeForces/python/B. Xenia and Ringroad.py
host = 1
time_unit = 0
input1 = input().split(" ")
n = int(input1[0]) 
m = int(input1[1]) 

input2 = input().split(" ")
tasks = []
for i in range(m):
    tasks.append(int(input2[i]))

while (True):
    if tasks[0] == host:
        while (True):
            if (len(tasks) > 1 and tasks[0] == tasks[1]):
                tasks.remove(host)
            tasks.remove(host)
            break
        if len(tasks) == 0:
            break
    if host == n:
        i = 1
        host = 0
    host += 1
    time_unit += 1

print(time_unit)

File: CodeForces/cpp/B. Xenia and Ringroad.cpp

#include<iostream>
#include<vector>
using namespace std;
int main() {
	
	int host = 1;
	int time_unit = 0;
	int n, m;
	cin >> n >> m;
	vector <int>tasks;
	int t;
	for (int i = 0; i < m; i++) {
		cin >> t;
		tasks.push_back(t);
	}
	while (true) {
		
			if (tasks[0] == host) {
				while (true) {
					if (tasks.size() > 1 && tasks[0] == tasks[1]) {
						tasks.erase(tasks.begin());
					}
					tasks.erase(tasks.begin());
					break;
				}
			}
			if (tasks.empty() == true) { break; }
			if (host == n) {
				host = 0;
			}
			time_unit += 1;
			host += 1;
	}

	cout << time_unit;
	
}
